src/Akka.CQRS.Hosting.Tests/TradeProcessor.cs

```diff
@@ -0,0 +1,295 @@
+        public TradeProcessor(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+        private async Task<IHost> TraderHost()
+        {
+            var tcs = new TaskCompletionSource();
+            using var cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(120));
+
+            var host = new HostBuilder()
+                .ConfigureServices(collection =>
+                {
+                    collection.AddAkka("AkkaTrader", builder  =>
+                    {
+                        #region Trade Processors
+                        builder
+                            .WithRemoting(new RemoteOptions
+                            {
+                                HostName = "127.0.0.1",
+                                PublicHostName = "127.0.0.1",
+                                Port = 5055,
+                                PublicPort = 5055,
+                            })
+                            .WithClustering(new ClusterOptions
+                            {
+                                SeedNodes = new[] { "akka.tcp://AkkaTrader@127.0.0.1:5055" },
+                                Roles = new[] { "trade-processor", "trade-events" }
+                            })
+                            .WithDistributedPubSub("trade-events")
+                            .WithShardRegion<OrderBookActor>("orderBook",
+                            (system, registry, resolver) => s => OrderBookActor.PropsFor(s),
+                            new StockShardMsgRouter(), new ShardOptions()
+                            {
+                                Role = "trade-processor"
+                            })
+                            .WithSqlServerPersistence(_sqlConnectionString, journalBuilder: builder =>
+                            {
+                                builder.AddWriteEventAdapter<StockEventTagger>("stock-tagger", new[] { typeof(IWithStockId) });
+                            })
+                            .AddHocon(@$"akka.persistence.journal.sql.provider-name = SqlServer.2019", HoconAddMode.Prepend)
+                            .AddPetabridgeCmd(cmd =>
+                            {
+                                _output.WriteLine("   PetabridgeCmd Added");
+                                cmd.RegisterCommandPalette(ClusterCommands.Instance);
+                                cmd.RegisterCommandPalette(ClusterShardingCommands.Instance);
+                                cmd.RegisterCommandPalette(new RemoteCommands());
+                                cmd.Start();
+                            });
+                        #endregion
+                    });
+                }).Build();
+
+            await host.StartAsync(cancellationTokenSource.Token);
+            //await Task.Delay(TimeSpan.FromSeconds(120));
+            await (tcs.Task.WaitAsync(cancellationTokenSource.Token));
+
+            return host;
+        }
+        private async Task TraderHostProx()

```

Comment:
Rather than start 3 separate hosts to try to form a cluster - my suggestion is to run all actors for all services inside the same `ActorSystem`. We're here to test that actor + Akka.Persistence functionality works. We can assume that the cluster roles / formation stuff, provided that it is configured correctly, won't impact this (because: actor location transparency.)

Reply:
This step will be easier to implement if you follow my recommendation here: https://github.com/Aaronontheweb/InMemoryCQRSReplication/pull/33/files#r1171418581